### PR TITLE
PS207-Protect-routes

### PR DESCRIPTION
**Pull request by mvstoyan**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -16,6 +16,13 @@ function App() {
   const { status } = useAuth();
   const isLoggedIn = status === AuthStatus.LoggedIn;
 
+  const ProtectedRoute = ({ children }) => {
+    if (!isLoggedIn) {
+      return <Navigate to="/signIn" />;
+    }
+    return children;
+  };
+
```
Comment (akosasante) [inline src/App.js]: Sorry this had slipped my mind when writing the ticket; but I belatedly realized we could have also done this with the `RequireAuth` component provided by `react-auth-context` but this works well as well; feel free to keep this or experiment with `RequireAuth` (though using RequireAuth may require some further changes to the auth settings in `index.js` so 🤷 )

https://github.com/akosasante/react-auth-context#:~:text=Alternatively%2C%20use%20the%20RequireAuth%20component%20in%20your%20router%20to%20wrap%20your%20component%20that%20should%20only%20be%20viewed%20if%20logged%2Din%3A

**Review by akosasante — APPROVED** [review]
LGTM!

File: src/App.js (hunk 1 above)
Comment (mvstoyan) [reply src/App.js]: I'm trying to do this option, but I think I'm going wrong

File: src/index.js [hunk 2]
```diff
@@ -37,11 +37,11 @@ root.render(
     {/* Use CSS baseline to set common spacing/sizing across all of our styling */}
     <CssBaseline />
     <BrowserRouter>
-      <AuthProvider {...authSettings}>
+      <RequireAuth {...authSettings}>
```
Comment (akosasante) [inline src/index.js]: This one should still be `AuthProvider`

`AuthProvider` is the "Context Provider" component that allows all the children of this element to have access to the useAuth hook. More about React Contexts here: https://react.dev/reference/react/createContext#provider

```suggestion
      <AuthProvider {...authSettings}>
```

File: src/index.js
Comment (akosasante) [inline src/index.js:1]: If using `RequireAuth`, then in this file we should make sure to update `loginPath` on line 22 to be the path on the frontend, where we want the user to be redirected to if the `RequireAuth` element finds that they are not logged in (probably `/signIn`).

File: src/App.js [hunk 3]
```diff
@@ -18,30 +19,147 @@ function App() {
...
-        <Route
-          path="/chat"
-          element={
-            <AlbumChat spotifyUrl="https://api.spotify.com/v1/albums/6r1lh7fHMB499vGKtIyJLy" />
-          }
-        />
-        {/* Below route is the catch-all route. It **MUST** be the last route because react-router checks the routes from top-to-bottom */}
-        <Route path="*" element={<PageNotFound />} />
-      </Routes>
+      <AuthProvider>
```
Comment (akosasante) [inline src/App.js]: Let's keep all the provider elements (like BrowserRouter which provides us with access to react-router hooks, and AuthProvider which provides us with access to auth hooks) in the `index.js` file; just for clearer separation of responsibiliites.

Comment (mvstoyan) [reply src/App.js]: Hope, I did right😅

File: src/App.js [hunk 4]
```diff
@@ -45,3 +76,88 @@ function App() {
...
+//           }
+//         />
+//         {/* Below route is the catch-all route. It **MUST** be the last route because react-router checks the routes from top-to-bottom */}
+//         <Route path="*" element={<PageNotFound />} />
+//       </Routes>
+//     </>
+//   );
+// }
+
+// export default App;
```
Comment (akosasante) [inline src/App.js]: 🗑️

**Review by akosasante — APPROVED** [review]
Looks great 🚢 !

**Pull request merged**